pkgs/top-level/python-packages.nix

```diff
@@ -11918,7 +11918,10 @@ in {
   pika-pool = callPackage ../development/python-modules/pika-pool { };
   platformio = callPackage ../development/python-modules/platformio { };
 
-  kmsxx = callPackage ../development/libraries/kmsxx { };
+  kmsxx = callPackage ../development/libraries/kmsxx {
+    # overrideCC isn't defined here, let's do without it
+    stdenv = stdenv.override { allowedRequisites = null; cc = pkgs.gcc6; };
```

Comment:
You can use
```
stdenv = pkgs.overrideCC stdenv pkgs.gcc6;
```

Reply:
However, I think doing it similar to dlib expression would be a better approach:
```
  dlib = buildPythonPackage rec {
    inherit (pkgs.dlib) name src nativeBuildInputs meta;

    buildInputs = pkgs.dlib.buildInputs ++ [ self.boost ];

    checkInputs = with self; [ pytest ];
  };
```

Reply:
Thank you. I tried your second suggestion but did not get it to work (the package is built using cmake, not setuptools), so let's go with your first suggestion instead.

Reply:
I came up with an even better solution: `inherit (pkgs.kmsxx) stdenv;`.
And you should maybe add `name = "${python.executable}-${pkgs.kmsxx.name}"`.
So we have
```
kmsxx = (callPackage ../development/libraries/kmsxx {
  inherit (pkgs.kmsxx) stdenv;
}).overrideAttrs (oldAttrs: {
  name = "${python.executable}-${pkgs.kmsxx.name}";
});
```

Reply:
Nice! Works fine, but I think we should use `name = "${python.libPrefix}-${pkgs.kmsxx.name}"`, otherwise the python3 package will be called `python3.6m-kmsxx...`